tools/structure/src/main.py

```diff
@@ -47,21 +48,25 @@ def run(
 
         self.stream_log("Indexing document...")
         file_hash = self.get_exec_metadata.get(MetadataKey.SOURCE_HASH)
-        tool_index = ToolIndex(tool=self)
+        index = Index(tool=self)
         tool_id = tool_metadata[SettingsKeys.TOOL_ID]
         outputs = tool_metadata[SettingsKeys.OUTPUTS]
         try:
+            usage_kwargs: dict[Any, Any] = dict()
+            run_id = CommonUtils.get_uuid()
```

Comment:
NIT: We could rename the function to `generate_uuid()` to be a bit more specific on what we're doing and eliminate doubts of whether its retrieved from somewhere

Reply:
Done